pkg/loader/openapi.go

```diff
@@ -115,6 +116,13 @@ func getOpenAPITools(t *openapi3.T, defaultHost string) ([]types.Tool, error) {
 				toolDesc = toolDesc[:1024]
 			}
 
+			toolName := operation.OperationID
+			if toolName == "" {
+				// When there is no operation ID, we use the method + path as the tool name and remove all characters
+				// except letters, numbers, underscores, and hyphens.
+				toolName = regexp.MustCompile(`[^a-zA-Z0-9_-]+`).ReplaceAllString(strings.ToLower(method)+strings.ReplaceAll(pathString, "/", "_"), "")
+			}
+
```

Comment:
nit: Not a blocker, but some autogold unit tests for `getOpenAPITools(...)` would make me much more comfortable with changes like this, and at first glance it doesn't look like they would be too much extra work to add.

Reply:
Great idea

Reply:
Added these